sdk/attestation/Azure.Security.Attestation/src/AttestationAdministrationClient.cs

```diff
@@ -140,22 +117,51 @@ public virtual AttestationResponse<string> GetPolicy(AttestationType attestation
+            => await GetPolicyInternalAsync(attestationType, true, cancellationToken).ConfigureAwait(false);
+
+        /// <summary>
+        /// Retrieves the attesttion policy for the specified <see cref="AttestationType"/>.
+        /// </summary>
+        /// <param name="attestationType"><see cref="AttestationType"/> to retrive.</param>
+        /// <param name="cancellationToken">Cancellation token used to cancel this operation.</param>
+        /// <param name="async">True if the call should be asynchronous.</param>
+        /// <returns>An <see cref="AttestationResponse{String}"/> with the policy for the specified attestation type.</returns>
+        /// <remarks>
+        /// This API returns the underlying attestation policy object stored in the attestation service for this <paramref name="attestationType"/>.
+        ///
+        /// The actual service response to the API is an RFC 7519 JSON Web Token. This token can be retrieved from <see cref="AttestationResponse{T}.Token"/>.
+        /// For the GetPolicy API, the body of the <see cref="AttestationResponse{T}.Token"/> is a <see cref="StoredAttestationPolicy"/> object, NOT a string.
+        /// </remarks>
+        private async Task<AttestationResponse<string>> GetPolicyInternalAsync(AttestationType attestationType, bool async, CancellationToken cancellationToken = default)
         {
             using DiagnosticScope scope = _clientDiagnostics.CreateScope($"{nameof(AttestationAdministrationClient)}.{nameof(GetPolicy)}");
             scope.Start();
             try
             {
-                var result = await _policyClient.GetAsync(attestationType, cancellationToken).ConfigureAwait(false);
-                var token = new AttestationToken(result.Value.Token);
+                Response<PolicyResponse> result;
+                if (async)
+                {
+                    result = await _policyClient.GetAsync(attestationType, cancellationToken).ConfigureAwait(false);
+                }
+                else
+                {
+                    result = _policyClient.Get(attestationType, cancellationToken);
+                }
+
+                var token = AttestationToken.Deserialize(result.Value.Token, _clientDiagnostics);
                 if (_options.TokenOptions.ValidateToken)
                 {
-                    await token.ValidateTokenAsync(_options.TokenOptions, await GetSignersAsync(cancellationToken).ConfigureAwait(false), cancellationToken).ConfigureAwait(false);
+                    if (!await token.ValidateTokenInternal(_options.TokenOptions, await GetSignersAsync(cancellationToken).ConfigureAwait(false), async, cancellationToken).ConfigureAwait(false))
+                    {
+                        throw new Azure.RequestFailedException("Attestation Token was rejected.");
+                    }
```

Comment:
We may want to consider using the `if (async)` block to split the token validation and querying the signers into the explicit sync/async calls, since both have synchronous versions.  I don't think `ValidateTokenInternal` is a problem, because it should honor the `async` parameter passed to it and return a completed task.   However, it looks like the `GetSignersAsync` call is a true async path and I think that potentially gets us into undesirable threadpool usage.

This pattern appears in a few places where tokens are validated; to avoid spam, I'll leave this stand as my only mention.

```suggestion
                    if (async)
                    {
                        if (!await token.ValidateTokenAsync(_options.TokenOptions, await GetSignersAsync(cancellationToken).ConfigureAwait(false), async, cancellationToken).ConfigureAwait(false))
                        {
                            throw new Azure.RequestFailedException("Attestation Token was rejected.");
                        }
                    }
                    else                    
                    {
                        if (!token.ValidateToken(_options.TokenOptions, GetSigners(cancellationToken), cancellationToken)
                        {
                            throw new Azure.RequestFailedException("Attestation Token was rejected.");
                        }
                    }
```

Reply:
Please feel free to seek a second opinion for that suggestion;  I'm mostly focused on a library that is asynchronous-only and may not be fully up-to-date on our current thinking for sync/async patterns.

Reply:
I'm not taking your suggestion, but the code *IS* broken. I've brought the async parameter into the GetSigners API and have switched to using that pattern throughout the code.